### PR TITLE
Document FoodComponent and it's builder

### PR DESCRIPTION
**Pull request by i509VCB**

### CODE REVIEW

File: mappings/net/minecraft/item/FoodComponent.mapping [hunk 1]
```diff
@@ -13,26 +14,58 @@ CLASS net/minecraft/class_4174 net/minecraft/item/FoodComponent
 		ARG 5 snack
 		ARG 6 statusEffects
 	METHOD method_19230 getHunger ()I
+		COMMENT Gets the amount of hunger a food item will fill.
+		COMMENT
+		COMMENT <p>One hunger is equivilant to half of a hunger bar icon.
+		COMMENT To fill a full hunger bar, this value must be at least 2.
```
Comment (liach) [inline mappings/net/minecraft/item/FoodComponent.mapping]:
```suggestion
```
Redundant.

File: mappings/net/minecraft/item/FoodComponent.mapping [hunk 2]
```diff
@@ -13,26 +14,58 @@ CLASS net/minecraft/class_4174 net/minecraft/item/FoodComponent
...
 	METHOD method_19231 getSaturationModifier ()F
+		COMMENT Gets the staturation modifier of a food item.
+		COMMENT
+		COMMENT <p>This value is typically used to determine how long a player can sustain the current hunger value before the hunger is used.
 	METHOD method_19232 isMeat ()Z
+		COMMENT Checks if a food item can be fed to dogs.
 	METHOD method_19233 isAlwaysEdible ()Z
+		COMMENT Checks a food item can be eaten regardless of the current hunger value.
```
Comment (liach) [inline mappings/net/minecraft/item/FoodComponent.mapping]:
```suggestion
		COMMENT Checks if a food item can be eaten when the current hunger bar is full.
```
Afaik in vanilla this limit is always "hunger value is full", and if there is no other restriction on hunger value we can just describe this specific case.

Ignore me if there can be other restrictions on hunger values aside from being not full.

File: mappings/net/minecraft/item/FoodComponent.mapping [hunk 3]
```diff
@@ -13,26 +14,58 @@ CLASS net/minecraft/class_4174 net/minecraft/item/FoodComponent
...
 	METHOD method_19234 isSnack ()Z
+		COMMENT Checks if a food item is snack-like and is eaten quickly.
 	METHOD method_19235 getStatusEffects ()Ljava/util/List;
+		COMMENT Gets a list of all status effect instances that may be applied when a food item is consumed.
+		COMMENT
+		COMMENT <p>The first value in the pair is the status effect instance to be applied.
+		COMMENT <p>The second value is a percentage change the status effect is applied on a scale between {@code 0.0F} and {@code 1.0F}
```
Comment (liach) [inline mappings/net/minecraft/item/FoodComponent.mapping]:
```suggestion
		COMMENT <p>The second value is the chance the status effect gets applied, on a scale between {@code 0.0F} and {@code 1.0F}.
```
Should be chance I think? Like 80% hunger for rotten flesh?

File: mappings/net/minecraft/item/FoodComponent.mapping [hunk 4]
```diff
@@ -13,26 +14,58 @@ CLASS net/minecraft/class_4174 net/minecraft/item/FoodComponent
...
 	CLASS class_4175 Builder
 		FIELD field_18620 hunger I
 		FIELD field_18621 saturationModifier F
 		FIELD field_18622 meat Z
 		FIELD field_18623 alwaysEdible Z
 		FIELD field_18624 snack Z
 		FIELD field_18625 statusEffects Ljava/util/List;
+		METHOD <init> ()V
+			COMMENT
```
Comment (liach) [inline mappings/net/minecraft/item/FoodComponent.mapping]:
```suggestion
```

File: mappings/net/minecraft/item/FoodComponent.mapping [hunk 5]
```diff
@@ -13,26 +14,58 @@ CLASS net/minecraft/class_4174 net/minecraft/item/FoodComponent
...
 		METHOD method_19236 meat ()Lnet/minecraft/class_4174$class_4175;
+			COMMENT Specifies that a food item can be fed to dogs.
 		METHOD method_19237 saturationModifier (F)Lnet/minecraft/class_4174$class_4175;
+			COMMENT Specifies the staturation modifier of a food item.
```
Comment (liach) [inline mappings/net/minecraft/item/FoodComponent.mapping]:
```suggestion
			COMMENT Specifies the saturation modifier of a food item.
```

File: mappings/net/minecraft/item/FoodComponent.mapping [hunk 6]
```diff
@@ -13,26 +14,58 @@ CLASS net/minecraft/class_4174 net/minecraft/item/FoodComponent
...
+			COMMENT
+			COMMENT <p>This value is typically used to determine how long a player can sustain the current hunger value before the hunger is used.
 			ARG 1 saturationModifier
+				COMMENT the saturation modifier
 		METHOD method_19238 hunger (I)Lnet/minecraft/class_4174$class_4175;
+			COMMENT Specifies the amount of hunger a food item will fill.
+			COMMENT
+			COMMENT <p>One hunger is equivilant to half of a hunger bar icon.
+			COMMENT To fill a full hunger bar, this value must be at least 2.
```
Comment (liach) [inline mappings/net/minecraft/item/FoodComponent.mapping]:
```suggestion
```

File: mappings/net/minecraft/item/FoodComponent.mapping [hunk 7]
```diff
@@ -13,26 +14,58 @@ CLASS net/minecraft/class_4174 net/minecraft/item/FoodComponent
...
 			ARG 1 hunger
+				COMMENT the amount of hunger
 		METHOD method_19239 statusEffect (Lnet/minecraft/class_1293;F)Lnet/minecraft/class_4174$class_4175;
+			COMMENT Specifies a status effect to apply to an entity when a food item is consumed.
+			COMMENT This method may be called multiple times to apply several status effects when food is consumed.
 			ARG 1 effect
+				COMMENT the effect instance to apply
 			ARG 2 chance
+				COMMENT the percentage change the status effect is applied. On a scale of {@code 0.0F} to {@code 1.0F}
```
Comment (liach) [inline mappings/net/minecraft/item/FoodComponent.mapping]:
```suggestion
				COMMENT the chance the status effect is applied, on a scale of {@code 0.0F} to {@code 1.0F}
```

File: mappings/net/minecraft/item/FoodComponent.mapping [hunk 8]
```diff
@@ -13,26 +14,58 @@ CLASS net/minecraft/class_4174 net/minecraft/item/FoodComponent
...
 		METHOD method_19240 alwaysEdible ()Lnet/minecraft/class_4174$class_4175;
+			COMMENT Specifies a food item can be eaten regardless of the current hunger value.
```
Comment (liach) [inline mappings/net/minecraft/item/FoodComponent.mapping]:
```suggestion
			COMMENT Specifies that a food item can be eaten when the current hunger bar is full.
```

**Review by liach — APPROVED** [review]
Nice!

File: mappings/net/minecraft/item/FoodComponent.mapping [hunk 9]
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/class_4174 net/minecraft/item/FoodComponent
+	COMMENT Represents the components that make up the properties a food item.
```
Comment (YanisBft) [inline mappings/net/minecraft/item/FoodComponent.mapping]:
```suggestion
	COMMENT Represents the components that make up the properties of a food item.
```

File: mappings/net/minecraft/item/FoodComponent.mapping [hunk 10]
```diff
@@ -13,11 +14,24 @@ CLASS net/minecraft/class_4174 net/minecraft/item/FoodComponent
 		ARG 5 snack
 		ARG 6 statusEffects
 	METHOD method_19230 getHunger ()I
+		COMMENT Gets the amount of hunger a food item will fill.
+		COMMENT
+		COMMENT <p>One hunger is equivilant to half of a hunger bar icon.
```
Comment (YanisBft) [inline mappings/net/minecraft/item/FoodComponent.mapping]:
```suggestion
		COMMENT <p>One hunger is equivalent to half of a hunger bar icon.
```

File: mappings/net/minecraft/item/FoodComponent.mapping [hunk 11]
```diff
@@ -26,13 +40,28 @@ CLASS net/minecraft/class_4174 net/minecraft/item/FoodComponent
...
 		METHOD method_19237 saturationModifier (F)Lnet/minecraft/class_4174$class_4175;
+			COMMENT Specifies the saturation modifier of a food item.
+			COMMENT
+			COMMENT <p>This value is typically used to determine how long a player can sustain the current hunger value before the hunger is used.
 			ARG 1 saturationModifier
+				COMMENT the saturation modifier
 		METHOD method_19238 hunger (I)Lnet/minecraft/class_4174$class_4175;
+			COMMENT Specifies the amount of hunger a food item will fill.
+			COMMENT
+			COMMENT <p>One hunger is equivilant to half of a hunger bar icon.
```
Comment (YanisBft) [inline mappings/net/minecraft/item/FoodComponent.mapping]:
```suggestion
			COMMENT <p>One hunger is equivalent to half of a hunger bar icon.
```

**Review by YanisBft — APPROVED** [review]

**Review by Gegy — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**